tests/Mantis/ConfigParserTest.php

```diff
@@ -0,0 +1,432 @@
+		$this->addScalarCase( 'false', PHPUnit_Type::TYPE_BOOL );
+		$this->addScalarCase( 'true', PHPUnit_Type::TYPE_BOOL );
+
+		# Constants
+		$this->addScalarCase( 'VERSION_ALL', PHPUnit_Type::TYPE_NULL );         # null
+		$this->addScalarCase( 'VERSION_FUTURE', PHPUnit_Type::TYPE_BOOL );      # false
+		$this->addScalarCase( 'VERSION_RELEASED', PHPUnit_Type::TYPE_BOOL );    # true
+		$this->addScalarCase( 'OFF', PHPUnit_Type::TYPE_INT );                  # 0
+		$this->addScalarCase( 'DEVELOPER', PHPUnit_Type::TYPE_INT );            # int
+		$this->addScalarCase( " DEVELOPER\n", PHPUnit_Type::TYPE_INT );
+		$this->addScalarCase( 'MANTIS_VERSION', PHPUnit_Type::TYPE_STRING );    #string
+		$this->addScalarCase( " MANTIS_VERSION\n", PHPUnit_Type::TYPE_STRING );
+	}
+	
+	/**
+	 * Initialize the array test cases list
+	 */
+	private function initArrayTestCases() {
+		/**
+		 * Template for new test cases
+		 * ---------------------------
+		# comment
+		$this->addArrayCase(
+<<<'EOT'
+EOT
+ 		);
+		 * ---------------------------
+ */
+
+		/**
+		 * Simple arrays
+		 */
+
+		# empty
+		$this->addArrayCase( "array( )" );
+
+		# one element
+		$this->addArrayCase( "array( 1 )" );
+
+		# several elements, trailing delimiter
+		$this->addArrayCase( "array( 1, 2, )" );
+
+		# formatted whitespace
+		$this->addArrayCase( "array ( 1, 2, 3 )" );
+
+		# no whitespace
+		$this->addArrayCase( "array(1,2,3)" );
+
+		# arbitrary whitespace
+		$this->addArrayCase( "  array(\n1,\t2  ,    3 )\r  " );
+
+		# mixed types, quotes
+		$this->addArrayCase(
+<<<'EOT'
+array( 1, 'a', "b" )
+EOT
+		);
+
+		# nested quotes
+		$this->addArrayCase(
+<<<'EOT'
+array( '"a""b"""', "'a''b'''" )
+EOT
+		);
+
+		/**
+		 * Associative arrays
+		 */
+
+		# associative
+		$this->addArrayCase( "array( 0 => 'a', 1 => 'b' )" );
+
+		# associative, unordered keys
+		$this->addArrayCase( "array( 5 => 'a', 2 => 'b' )" );
+
+		# associative, text keys
+		$this->addArrayCase( "array( 'i' => 'a', 'j' => 'b' )" );
+
+		# associative, mixed keys
+		$this->addArrayCase( "array( 'i' => 'a', 1 => 'b', 'j' => 'c', 7 => 'd' )" );
+
+		# mixed associative, omitting some keys
+		$this->addArrayCase( "array( 'i' => 'a', 1 => 'b', 'c', 'j' => 'd' )" );
+
+		# mixed associative, overwriting implicit keys
+		$this->addArrayCase( "array( 0 => 'a0', 1 => 'a1', 'axx', 2 => 'a2' )" );
+
+		$this->addArrayCase(
+<<<'EOT'
+array(
+	array ( 1, 'a', 3 => 1, 4 => 'b', 'x' => 'y' )
+)
+EOT
+		);
+
+		/**
+		 * Use of constants
+		 */
+
+		# As array values (e.g. handle_bug_threshold)
+		$this->addArrayCase( "array( DEVELOPER, MANAGER )" );
+
+		# As array keys (e.g. status_enum_workflow)
+		$this->addArrayCase(
+<<<'EOT'
+array (
+  NEW_ => '20:feedback,30:acknowledged',
+  ACKNOWLEDGED => '40:confirmed',
+)
+EOT
+		);
+
+		# both (e.g. set_status_threshold)
+		$this->addArrayCase( 'array( NEW_ => REPORTER )' );
+
+		/**
+		 * Multidimensional arrays
+		 */
+
+		# notify_flags sample
```

Comment:
You already have it, can you add two elements on the top level to make sure all is OK.